packages/vite/src/node/server/index.ts

```diff
@@ -467,6 +467,17 @@ export async function _createServer(
         getDepsOptimizer(server.config, true)?.close(),
         closeHttpServer(),
       ])
+      // Await pending requests. We throw early in transformRequest
+      // and in hooks if the server is closing, so the import analysis
+      // plugin stops pre-transforming static imports and this block
+      // is resolved sooner.
+      while (server._pendingRequests.size > 0) {
+        await Promise.allSettled(
+          [...server._pendingRequests.values()].map(
+            (pending) => pending.request,
+          ),
+        )
+      }
```

Comment:
I have an itch to refactor this code as:

```suggestion
      while (server._pendingRequests.size > 0) {
        await server._pendingRequests.values().next().request
      }
```

since the `_pendingRequests` are already running in parallel, so we avoid creating an array, and calling `allSettled`. Just a nit and a fun thought, so resolving this.

Reply:
That's interesting! It would force us to add a `try/catch` block to get the same as `allSettled`. I don't know if it would be that much more performant, so leaving it for now.

Reply:
I think we can chain a `.catch` too (though I've seen `.catch` to be slower than `try..catch` in some case somehow). But yeah it's not a big deal in this part of the code for now.